pkg/terraform/controller.go

```diff
@@ -28,6 +29,33 @@ func Register(
 	serviceAccounts corev1.ServiceAccountController,
 	jobs batchv1.JobController,
 ) {
+	// watch for modules
+	relatedresource.Watch(ctx, "state-module-watch",
+		func(namespace, name string, obj runtime.Object) ([]relatedresource.Key, error) {
```

Comment:
For better readability would be the best take out watcher implementation from argument:
```suggestion
	relatedresource.Watch(ctx, "state-module-watch",yourWatcherFunc,states, modules)
	relatedresource.Watch(ctx, "state-config-secret-watch",configAndSecretsFun, states, modules)
```
But I see they use some coding style:  Depends on you if you want to keep original coding-style or take out implementation into standalone functions (refactor both watchers needed).


LGTM

Reply:
@kuritka implementations are mine (part of upstream PR though). Yes. I have this in mind, will refactor later